jarviscli/plugins/news.py

```diff
@@ -1,224 +1,157 @@
-            self.is_news_site_configured = True
-
-        print("List of configured news sites are as below")
-
-        for idx, site in enumerate(news_site_list):
-            print("{} : {}".format(idx + 1, site))
+            jarvis.say(news_source + " was already included in your sources!", Fore.GREEN)
+        return self.get_news_sources(jarvis)
 
-        print("Would you like to delete any in the list?(y/n)")
-        del_from_list = str(input())
-        if del_from_list.lower() == "y":
-            print("Enter the site index to be deleted. If multiple sites, then enter index with comma seperated values")
-            index_to_del = str(input())
-            items_idx_to_del = index_to_del.split(',')
-            list_to_del = [news_site_list[int(item) - 1] for item in items_idx_to_del]
-            news_site_list = [item for item in news_site_list if item not in list_to_del]
-            if len(news_site_list) == 0:
-                self.is_news_site_configured = False
-        jarvis.update_data(News.get_key(), news_site_list)
-        self.news_options(jarvis)
-        self.get_news()
-
-    def quick_news(self):
-        self.request_news()
-    '''
-        Gets and returns JSON data of news
-    '''
-
-    def get_news_json(self):
-        try:
-            response = urllib.request.urlopen(self.url)
-        except AttributeError:
-            response = urllib.urlopen(self.url)
-        return json.loads(response.read().decode('utf-8'))
-
-    '''
-        This sets the users options and loads them from Memory
-        if they exist
-    '''
+    """
+        removes a new source from the news channel of the user
+    """
+    def remove_source(self, jarvis, news_source):
+        sources = self.get_news_sources(jarvis)
+        if news_source in sources:
+            sources.remove(news_source)
+            jarvis.update_data("news-sources", sources)
+            jarvis.say(news_source + " has been successfully been removed from your sources!", Fore.GREEN)
+        return self.get_news_sources(jarvis)
 
-    def news_options(self, jarvis):
-        # check to see if user already has default news source
-        if jarvis.get_data('news-source'):
-            print("your default news source is {}"
-                  .format(jarvis.get_data('news-source')))
-            print("Would you like news from this source? (yes/no): ")
-            x = input()
-            if x == 'y' or x == 'yes':
-                self.source = jarvis.get_data('news-source')
-            # if news site configured already, display it
-            elif self.is_news_site_configured:
-                get_configured_opt(jarvis)
-            # if not set get users preference
-            else:
-                self.get_default_opt(jarvis)
-        elif self.is_news_site_configured:
-            self.get_configured_opt(jarvis)
+    """
+        configures the news channel of the user
+    """
+    def configure(self, jarvis):
```

Comment:
In configure function, IMO there is no need of choice if a user has entered `news configure`. Maybe give an index to not make any changes